File: contrib/java/ManyMouse.java
/*
 * Java bindings to the ManyMouse C code, via JNI.
 *
 * Please see the file LICENSE.txt in the source's root directory.
 *
 *  This file written by Ryan C. Gordon.
 */

public class ManyMouse
{
    // Native method hooks.
    public native static synchronized int Init();
    public native static synchronized void Quit();
    public native static synchronized String DeviceName(int index);
    public native static synchronized boolean PollEvent(ManyMouseEvent event);

    // JNI link.
    static { System.loadLibrary("ManyMouse"); }
} // ManyMouse

// end of ManyMouse.java ...

File: contrib/matlab/ManyMouse.m
classdef ManyMouse

    properties
    end
    
    methods (Static)
        
      
        function availableMice = init(  )
            availableMice = manymouse_mex( 'ManyMouse_Init' );
        end
        
        function quit( )
            manymouse_mex( 'ManyMouse_Quit' );
        end
        
        function sDriverName = driverName(  )
            sDriverName = manymouse_mex( 'ManyMouse_DriverName' );
        end
        
        function sDeviceNames = deviceName(  )
            sDeviceNames = manymouse_mex( 'ManyMouse_DeviceName' );
        end
        
        function event = pollEvent(  )
            event = manymouse_mex( 'ManyMouse_PollEvent' );
        end
        
        
    end
    
end

